web_monitoring/tests/test_db.py

```diff
@@ -0,0 +1,240 @@
+db_vcr = vcr.VCR(
+         serializer='json',
+         cassette_library_dir=cassette_library_dir,
+         record_mode='once',
+         match_on=['uri', 'method'],
+)
+global_stash = {}  # used to pass info between tests
+
+
+# Refers to real data that is part of the 'seed' dataset in web-monitoring-db
+URL = 'https://www3.epa.gov/climatechange/impacts/society.html'
+SITE = 'EPA - www3.epa.gov'
+AGENCY = 'EPA'
+PAGE_ID = '6c880bdd-c7a6-4bbf-a574-7d6479cc4fe8'
+TO_VERSION_ID = '9342c121-cff0-4454-934f-d0f118508da1'
+VERSIONISTA_ID = '10329339'
+
+# This is used in new Versions that we add.
+TIME = datetime(2017, 11, 15, tzinfo=timezone.utc)
+NEW_VERSION_ID = '06620776-d347-4abd-a423-a871620299a9'
+
+# The only matters when re-recording the tests for vcr.
+AUTH = {'url': "http://localhost:3000",
+        'email': "seed-admin@example.com",
+        'password': "PASSWORD"}
+
+
+def test_missing_creds():
+    try:
+        env = os.environ.copy()
+        os.environ.clear()
+        with pytest.raises(MissingCredentials):
+            Client.from_env()
+        os.environ.update({'WEB_MONITORING_DB_URL': AUTH['url'],
+                           'WEB_MONITORING_DB_EMAIL': AUTH['email'],
+                           'WEB_MONITORING_DB_PASSWORD': AUTH['password']})
+        Client.from_env()  # should work
+    finally:
+        os.environ.update(env)
+
+
+@db_vcr.use_cassette()
+def test_list_pages():
+    cli = Client(**AUTH)
+    res = cli.list_pages()
+    assert res['data']
+
+    # Test chunk query parameters.
+    res = cli.list_pages(chunk_size=2)
+    assert len(res['data']) == 2
+    res = cli.list_pages(chunk_size=5)
+    assert len(res['data']) == 5
+
+    # Test filtering query parameters.
+    res = cli.list_pages(url='__nonexistent__')
+    assert len(res['data']) == 0
+    res = cli.list_pages(url=URL)
+    assert len(res['data']) > 0
+    res = cli.list_pages(site='__nonexistent__')
+    assert len(res['data']) == 0
+    res = cli.list_pages(site=SITE)
+    assert len(res['data']) > 0
+    res = cli.list_pages(agency='__nonexistent__')
+    assert len(res['data']) == 0
+    res = cli.list_pages(agency=AGENCY)
+    assert len(res['data']) > 0
+
+
+@db_vcr.use_cassette()
+def test_get_page():
+    cli = Client(**AUTH)
+    res = cli.get_page(PAGE_ID)
+    assert res['data']['uuid'] == PAGE_ID
+
+
+@db_vcr.use_cassette()
+def test_list_page_versions():
+    cli = Client(**AUTH)
+    res = cli.list_versions(page_id=PAGE_ID)
+    assert all([v['page_uuid'] == PAGE_ID for v in res['data']])
+
+
+@db_vcr.use_cassette()
+def test_list_versions():
+    cli = Client(**AUTH)
+    res = cli.list_versions()
+    assert res['data']
+
+
+@db_vcr.use_cassette()
+def test_get_version():
+    cli = Client(**AUTH)
+    res = cli.get_version(TO_VERSION_ID)
+    assert res['data']['uuid'] == TO_VERSION_ID
+    assert res['data']['page_uuid'] == PAGE_ID
+
+
+@db_vcr.use_cassette()
+def test_get_version_by_versionista_id():
```

Comment:
I'm not sure how this test passed.
If you use the API to get the data for the versionista id, you'll see that while other details match, the `uuid` and `page_uuid` are different.

Reply:
I think the data in the cassette for this test is incorrect. I could be wrong, still diagnosing this issue.

Reply:
I ran this against a local deployment using the seed data, not the staging deployment. I'll add a comment to make that clear.

Reply:
Should all tests have data from the same deployment?
I'm not familiar with the use of vcr, so I'm still figuring this part out.

Reply:
> Should all tests have data from the same deployment?

No. Data in the tests should not need to bear any relationship whatsoever to the data in any given deployment. It shouldn’t matter where it came from. We only care that the data is *valid* (not the same as production) and its *structure* matches that of the latest DB version.

Actual data from *any* DB (so long as its code is up-to-date) will satisfy the above conditions. VCR just records a request so it can be played back, meaning it will cover the above conditions without someone having to hand-code fixture data.